engine/classes/Elgg/Database/AccessCollections.php

```diff
@@ -45,38 +109,38 @@ public function __construct($site_guid) {
 	 */
 	function getAccessList($user_guid = 0, $site_guid = 0, $flush = false) {
 		global $init_finished;
-		$cache = _elgg_services()->accessCache;
-		
+		$cache = $this->access_cache;
+
 		if ($flush) {
 			$cache->clear();
 		}
-	
+
 		if ($user_guid == 0) {
-			$user_guid = _elgg_services()->session->getLoggedInUserGuid();
+			$user_guid = $this->session->getLoggedInUserGuid();
```

Comment:
This is improvement but the test (`ElggCoreAccessSQLTest::testLoggedOutUser`) is not prepared. It only changes the session in the service provider. It would need to inject the new session into this instance, or rebuild the instance. I'll work on it.